doonut-app-external-api/src/main/kotlin/com/doonutmate/oauth/kakao/service/KakaoOauthLoginService.kt

```diff
@@ -0,0 +1,35 @@
+package com.doonutmate.oauth.kakao.service
+
+import com.doonutmate.doonut.member.model.Member
+import com.doonutmate.doonut.member.model.OauthType
+import com.doonutmate.doonut.member.service.MemberBusinessService
+import com.doonutmate.oauth.kakao.dto.KakaoTokenRequest
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional
+class KakaoOauthLoginService(
+    private val kakaoAccessClientLoginService: KakaoAccessClientLoginService,
+    private val memberBusinessService: MemberBusinessService,
+) {
+
+    fun kakaoLoginPreventDuplication(tokenRequest: KakaoTokenRequest, oauthType: OauthType): Member? {
+        val savedId = kakaoAccessClientLoginService.getKakaoUserId(tokenRequest)
+        return memberBusinessService.getByOauthId(savedId.findMemberId())
+            ?: kakaoSignUpNewMember(tokenRequest, oauthType)
+    }
+
+    fun kakaoSignUpNewMember(tokenRequest: KakaoTokenRequest, oauthType: OauthType): Member? {
```

Comment:
여기도 ?을 제거해도 좋을 것 같아요.  
```suggestion
    fun kakaoSignUpNewMember(tokenRequest: KakaoTokenRequest, oauthType: OauthType): Member {
```